src/differentiation/jaches_products.jl

```diff
@@ -110,6 +111,7 @@ function numauto_hesvec!(dy,
     g(cache1, x)
     @. x -= 2ϵ * v
     g(cache2, x)
+    @. x += ϵ * v
```

Comment:
@ChrisRackauckas that this was not before seems to actually have been a fairly serious numerical error: the vector x where the Jacobian is being evaluated at gets edited by around sqrt(eps) each time the Jacobian is applied, if I am thinking this through correctly

Reply:
sorry, not Jacobian; this error was only present for (finite difference computed) Hessians